### PR TITLE
January campaign: Navigate to the popup

### LINKED ISSUE #6616: January 2020 Campaign Popup Plumbing and VM
## 🎯 What needs to be done?
We want to add a relatively simple popup for our upcoming winter discount campaign.

This issue deals with the shared remote config, view model and metrics.

## 📝 Detailed explanation

Showing the popup:
- The popup will be disabled by default (we will enable it using a firebase remote config parameter).
- The popup will show when the main view is shown (be that on a fresh start, activation from background, or navigation from another screen in-app), **if the user has installed the app more than 48 hours ago** and **if the user has at least two time entries in the local database**.
- As soon as the popup is shown, we set a flag in shared storage that will prevent it from ever being shown again.
- There is no need to close the popup when the user leaves the app with it open (i.e. it should still be open when the user comes back quickly). However, if the app gets cleaned up in the background, we also don't have to restore the popup if it was open. I.e. we're keeping it as simple as possible.

The popup itself:
- There are two slightly different versions (Option A and Option B), with slightly different copy and a slightly different link. Which option to use will be part of the remote config parameter.
- The popup will have various images and/or text specified in platform-specific follow-up issues.
- The popup will have a call-to-action button which, when tapped, will close the popup and open a URL (specified in platform-specific issues) in the browser (NOT in the in-app browser!) 
- The popup will have a dismiss button that closes it.

Events:
- A. One when the popup is shown (with a parameter indicating what campaign this is)
- B. One when the call-to-action is hit (with a parameter indicating what campaign this is)
- C. One when the dismiss button is hit (with a parameter indicating what campaign this is)

Note that if correctly implemented, B. and C. are mutually exclusive, and B+C<=A (< because some users will close the app and lose the popup that way).

## 🎨 Graphical resources
Provided in follow up platform-specific issues.

## 🛠 Technical remarks
The firebase remote config flag will be called `January2020CampaignOption`. It defaults to `none` and can be `A` or `B` which will cause the correct option of the popup to be shown.

The shared-storage parameter will be called "ShowedJanuary2020Campaign" (I'm not sure about the canonical formatting here, feel free to change that if camel or snake case are most customary). It defaults to `false` and is set to `true` when the popup is shown. 

The new events are called:
- A. `MarketingMessageShown` with property `Campaign="january2020"`
- B. `MarketingMessageCallToActionHit` with property `Campaign="january2020"`
- C. `MarketingMessageDismissed` with property `Campaign="january2020"`

Despite the property values currently being hardcoded, this should be done in the view model, not in the event itself. We'll likely use the same event in the future for other marketing campaigns.

## 🏆 Outcome

- [x] Added remote config property, including default value, and expose it in relevant service
- [x] Added shared storage property and expose it in relevant service
- [x] Added view model with call to action and dismiss functions
- [x] Integrated view model to be shown on the correct conditions on the main view (will this work without actual views? Either way the condition to show the popup should be part of this issue)
- [x] Added all three events and called them at the appropriate points

### PR DESCRIPTION
**Pull request by simonrozsival**
## What's this?
This PR implements navigation to the Campaign VM if some conditions are met.

### Relationships
Ref #6616, Requires #6643 

## How is it done?
Most of the code changes is just caused by injecting dependencies into the `MainViewModel`.

The `shouldShowJanuary2020Campaign` mirrors the business logic described in the issue. I had some follow-up issues and so this logic might not be final.

## Review hints
Check the specs in the issue + my comments in the issue thread.

## :squid: Permissions
🦑

### CODE REVIEW

File: Toggl.Core.UI/ViewModels/MainViewModel.cs [hunk 1]
```diff
@@ -391,6 +402,22 @@ private async Task handleNoDefaultWorkspaceState()
             }
         }
 
+        private async Task<bool> shouldShowJanuary2020Campaign()
+        {
+            var isDisabled = remoteConfigService.GetJanuary2020CampaignConfiguration().Option == January2020CampaignConfiguration.AvailableOption.None;
+            if (isDisabled) return false;
```
Comment (daividssilverio) [inline Toggl.Core.UI/ViewModels/MainViewModel.cs]: No early returns on single lines if the condition has a return value.

File: Toggl.Core.UI/ViewModels/MainViewModel.cs [hunk 2]
```diff
@@ -391,6 +402,22 @@ private async Task handleNoDefaultWorkspaceState()
...
+
+            var wasAlreadyShown = OnboardingStorage.WasJanuary2020CampaignShown();
+            if (wasAlreadyShown) return false;
```
Comment (daividssilverio) [inline Toggl.Core.UI/ViewModels/MainViewModel.cs]: same here.

File: Toggl.Core.UI/ViewModels/MainViewModel.cs [hunk 3]
```diff
@@ -391,6 +402,22 @@ private async Task handleNoDefaultWorkspaceState()
...
+
+            var isNewUser =
```
Comment (daividssilverio) [inline Toggl.Core.UI/ViewModels/MainViewModel.cs]: This chain of conditions does not define a new user, right? Only an user that has logged in more than 48 hours ago and has at last one time entry.

Comment (simonrozsival) [reply Toggl.Core.UI/ViewModels/MainViewModel.cs]: I'll change it to some better name.

**Comment by daividssilverio** [comment]
Dismissing my request for changes as they were met.

**Comment by simonrozsival** [comment]
I added unit tests. They are ugly, but they test exactly what they should. I don't want to clean up the code, because this code will be removed in January anyway. Let's get this over with.

**Pull request merged**